packages/fxa-profile-server/lib/routes/ecosystem_anon_id/post.js

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Joi = require('@hapi/joi');
+
+const logger = require('../logging')('routes.ecosystem_anon_id.post');
+const notifyProfileUpdated = require('../../updates-queue');
+
+module.exports = {
+  auth: {
+    strategy: 'oauth',
+    scope: ['profile:ecosystem_anon_id:write'],
+  },
+  validate: {
+    payload: {
+      ecosystemAnonId: Joi.string().required(),
+    },
+  },
+  handler: async function ecosystemAnonIdPost(req) {
```

Comment:
Ideally, clients would be able to `If-Match` or `If-Unmodified-Since` on this post, for the case when we're filling in a placeholder anon id and we don't want two clients to race in setting it.

Reply:
Sounds good. I've added a note to https://github.com/mozilla/fxa/issues/4325, since I can't test the ETag is checked against the If-Match value until the DB returns an actual response.